solr/modules/hdfs/src/java/org/apache/solr/hdfs/snapshots/PrepareSnapshotExportCLI.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.hdfs.snapshots;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.hadoop.fs.Path;
+import org.apache.solr.cli.CLIO;
+import org.apache.solr.cli.SolrCLI;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.response.CollectionAdminResponse;
+import org.apache.solr.common.cloud.DocCollection;
+import org.apache.solr.common.cloud.Replica;
+import org.apache.solr.common.cloud.Slice;
+import org.apache.solr.common.params.CollectionAdminParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.snapshots.CollectionSnapshotMetaData;
+import org.apache.solr.core.snapshots.CollectionSnapshotMetaData.CoreSnapshotMetaData;
+import org.apache.solr.core.snapshots.SolrSnapshotManager;
+
+/**
+ * This class provides the commandline to prepare a snapshot export with HDFS. Other snapshot
+ * related commands live in org.apache.solr.cli package.
+ */
+public class PrepareSnapshotExportCLI implements Closeable, CLIO {
+
+  private static final String COLLECTION = "c";
+  private static final String TEMP_DIR = "t";
+  private static final String DEST_DIR = "d";
+  private static final String HDFS_PATH_PREFIX = "p";
+  private static final String BACKUP_REPO_NAME = "r";
+  private static final String ASYNC_REQ_ID = "i";
```

Comment:
Would also be nice to support the long form of the options, so that it is consistent with the other scripts.

May also be a separate ticket though.

Reply:
now that I've deleted `PrepareSnapshotExportCLI` this no longer applies.   HOwever, *maybe* we would want to look at `SolrSnapshotsTool` remaining methods (once the deprecated ones are removed in 10) and look at long/short thing...!